client/src/index.js

```diff
@@ -1,19 +1,37 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import './index.css';
+// import jwt_decode from 'jwt-decode';
+// import * as APIUtil from './util/session_api_util';
+import configureStore from './store/store';
+import App from './App.jsx';
 import * as serviceWorker from './serviceWorker';
-import {IndexPage, Locations, Barbers, About} from './Pages';
-import {Router} from '@reach/router'
-ReactDOM.render(
-<Router>
-<IndexPage path="/" />   
-<Locations path="/locations" />
-<Barbers path="/barbers" /> 
-<About path="/about" />
-</Router> 
-    , document.getElementById('root'));
 
+document.addEventListener('DOMContentLoaded', () => {
+  let store = configureStore();
+  // for front end auth, comment or replace commented out code below
+  // // Check for token
+  // if (localStorage.jwtToken) {
+  //   // Set auth token header auth
+  //   APIUtil.setAuthToken(localStorage.jwtToken);
+  //   // Decode token and get user info and exp
+  //   const decoded = jwt_decode(localStorage.jwtToken);
+  //   // Set user and isAuthenticated
+  //   store.dispatch(APIUtil.setCurrentUser(decoded));
+  //
+  //   // Check for expired token
+  //   const currentTime = Date.now() / 1000;
+  //   if (decoded.exp < currentTime) {
+  //     // Logout user
+  //     store.dispatch(APIUtil.logoutUser());
+  //     // Redirect to login
+  //     window.location.href = '/login';
+  //   }
+  // }
+  const root = document.getElementById('root');
+  ReactDOM.render(<App store={store} />, root);
```

Comment:
@UjwalBattar this one right here.

Reply:
And here, we are actually creating and configuring the app store when the app initially loads

Reply:
ohhh. ok your just passing the store as a props. cool. ok man. this is good. ill merge this.